### PR TITLE
Add ability to reorder folds

### PR DESCRIPTION
**Pull request by VeraZab**
closes: #717

### CODE REVIEW

**Comment by VeraZab** [comment]
@nicolaskruchten I think this is what we want would you check please ✨ 
I didn't add it to other accordions, but this easily could be extended to them

**Comment by VeraZab** [comment]
@dmt0 I think this is ready for code review as well

**Comment by VeraZab** [comment]
adjusted styles
Screen Shot 2019-04-10 at 11 02 25 AM

**Comment by VeraZab** [comment]
so I've added the ability to reorder: images, shapes, annotations on top of traces

for transforms, I wasn't sure about the array to reorder (it changes based on the transform applied, and I'm not sure of the side effects) so I didn't do it here.

for rangeselector buttons, the path to get to the array to reorder is a bit less straight forward. I could perhaps use the `nested property` function of plotly.js. but didn't do it here.

just picked the low hanging fruit. let me know if that's ok @nicolaskruchten

File: src/components/containers/PlotlyFold.js [hunk 1]
```diff
@@ -72,8 +72,30 @@ export class Fold extends Component {
...
+      <div className="fold__top__moving-controls">
+        <span
+          className="fold__top__moving-controls--up"
+          onClick={e => {
+            // prevents fold toggle to happen when clicking on moving arrow controls
+            e.stopPropagation();
+            alert('up');
+          }}
+        >
+          {canMoveUp ? <AngleDownIcon /> : null}
```
Comment (dmt0) [inline src/components/containers/PlotlyFold.js]: would it make sense to move this check outside the `span`?

File: src/EditorControls.js [hunk 2]
```diff
@@ -289,6 +289,26 @@ class EditorControls extends Component {
         }
         break;
 
+      case EDITOR_ACTIONS.MOVE_TO:
+        // checking is fromIndex and toIndex is a number because just checking
+        // if not there will not work as index can be 0 and that value is falsy
```
Comment (dmt0) [inline src/EditorControls.js]: very hard to read this comment :)

File: src/EditorControls.js [hunk 3]
```diff
@@ -294,10 +294,10 @@ class EditorControls extends Component {
         // if not there will not work as index can be 0 and that value is falsy
         if (payload.path && !isNaN(payload.fromIndex) && !isNaN(payload.toIndex)) {
           if (payload.path === 'data') {
-            const a = graphDiv.data[payload.fromIndex];
-            const b = graphDiv.data[payload.toIndex];
-            graphDiv.data.splice(payload.toIndex, 1, a);
-            graphDiv.data.splice(payload.fromIndex, 1, b);
+            const traceMoved = graphDiv.data[payload.fromIndex];
+            const traceReplaced = graphDiv.data[payload.toIndex];
```
Comment (dmt0) [inline src/EditorControls.js]: `traceFrom`, `traceTo`? When you flip two traces, how do you decide which one is moved and which one is replaced?

File: src/components/containers/PlotlyFold.js (hunk 1 above)
Comment (dmt0) [reply src/components/containers/PlotlyFold.js]: nevermind...

**Comment by dmt0** [comment]
💃 !

File: src/EditorControls.js (hunk 3 above)
Comment (VeraZab) [reply src/EditorControls.js]: yup sorry, should have said that this one is not reviewable commit by commit..

**Comment by nicolaskruchten** [comment]
✨ although I don't love the "you can't click here" cursor on the greyed out arrows... I'd just leave it as a normal non-link cursor.

**Comment by VeraZab** [comment]
> although I don't love the "you can't click here" cursor on the greyed out arrows
yeah, agree, removed!

**Pull request merged**